src/WebJobs.Script/Rpc/FunctionRegistration/FunctionDispatcher.cs

```diff
@@ -138,13 +107,24 @@ internal void ShutdownWebhostLanguageWorkerChannels()
             _webHostLanguageWorkerChannelManager.ShutdownChannels();
         }
 
-        private void StartWorkerProcesses(int startIndex, Action startAction)
+        private async Task StartWorkerProcesses(int startIndex)
         {
+            // Start initializing first right away
+            if (startIndex == 0)
+            {
+                await InitializeLanguageWorkerChannel(0);
+                startIndex = 1;
+            }
+
+            // Offset start of others by _debounceSeconds
             for (var count = startIndex; count < _maxProcessCount; count++)
             {
+                Action startAction = async () => await InitializeLanguageWorkerChannel(0);
                 startAction = startAction.Debounce(_processStartCancellationToken.Token, count * _debounceSeconds * 1000);
                 startAction();
             }
+
+            State = FunctionDispatcherState.Initialized;
```

Comment:
Who listens for this state? Would it make sense to set it to `Initialized` after the first channel was set up? Or does `Debounce` make the call to `startAction()` fire-and-forget?